Tests/XCUITests/HistoryTests.swift

```diff
@@ -103,215 +98,214 @@ class HistoryTests: BaseTestCase {
-        navigator.openURL(path(forTestPage: "test-mozilla-book.html"))
-        waitUntilPageLoad()
-        waitForTabsButton()
-        navigator.goto(TabTray)
-        navigator.performAction(Action.AcceptRemovingAllTabs)
-        navigator.performAction(Action.CloseURLBarOpen)
-        navigator.nowAt(NewTabScreen)
-        navigator.goto(BrowserTabMenu)
+    func testRecentlyClosedWebsiteClosed() {
+        openBookOfMozilla()
+
+        closeFirstTabByX()
+
+        // On regular mode, the closed tab is listed in "Recently Closed Tabs List"
+        navigator.nowAt(HomePanelsScreen)
         navigator.goto(HistoryRecentlyClosed)
+        waitForExistence(app.tables["Recently Closed Tabs List"])
+        XCTAssertTrue(app.tables.cells.staticTexts[bookOfMozilla["label"]!].exists)
+    }
 
-        // The Closed Tabs list should contain the info of the website just closed
-        waitForExistence(app.tables["Recently Closed Tabs List"], timeout: 3)
-        XCTAssertTrue(app.tables.cells.staticTexts[closedWebPageLabel].exists)
+    func testRecentlyClosedPrivateMode() {
+        openBookOfMozilla()
 
-        navigator.goto(HomePanelsScreen)
+        closeFirstTabByX()
 
-        // This option should be enabled on private mode too
+        // Toggle to private mode. The closed tab is listed in "Recently Closed Tabs List"
+        navigator.nowAt(HomePanelsScreen)
         navigator.toggleOn(userState.isPrivate, withAction: Action.TogglePrivateMode)
-        navigator.performAction(Action.OpenNewTabFromTabTray)
-        navigator.nowAt(NewTabScreen)
-        navigator.performAction(Action.CloseURLBarOpen)
-        navigator.nowAt(NewTabScreen)
-        navigator.goto(BrowserTabMenu)
         navigator.goto(HistoryRecentlyClosed)
         waitForExistence(app.tables["Recently Closed Tabs List"])
+        XCTAssertTrue(app.tables.cells.staticTexts[bookOfMozilla["label"]!].exists)
     }
 
-    func testClearRecentlyClosedHistory() {
-        // Open the default website
-        navigator.openURL(path(forTestPage: "test-mozilla-book.html"))
-        navigator.nowAt(BrowserTab)
+    func testRemoveAllTabsButton() {
+        openBookOfMozilla()
+
+        // Tap "Remove All Tabs" instead of close the tab individually
         waitForTabsButton()
         navigator.goto(TabTray)
         navigator.performAction(Action.AcceptRemovingAllTabs)
-        navigator.performAction(Action.CloseURLBarOpen)
-        navigator.nowAt(NewTabScreen)
-        navigator.goto(BrowserTabMenu)
-        navigator.goto(HistoryRecentlyClosed)
-        // Once the website is visited and closed it will appear in Recently Closed Tabs list
-        waitForExistence(app.tables["Recently Closed Tabs List"])
-        XCTAssertTrue(app.tables.cells.staticTexts[closedWebPageLabel].exists)
 
-        navigator.goto(HomePanelsScreen)
-        navigator.nowAt(NewTabScreen)
+        // The closed tab is *not* listed in "Recently Closed Tabs List"
```

Comment:
I'm not sure if this is a bug or the design... I have asked @SimonBasca in case he knows, if not we should clarify with devs. I thought that using that button we did not see the test in the simulator via automated tests but I have checked manually and closing all tabs does not show them in the menu on my phone either.